src/imitation/rewards/reward_nets.py

```diff
@@ -163,78 +185,37 @@ def dtype(self) -> th.dtype:
             return th.get_default_dtype()
 
 
-class ShapedRewardNet(RewardNet):
-    """A RewardNet consisting of a base net and a potential shaping."""
+class RewardNetWrapper(RewardNet, abc.ABC):
+    """A RewardNet wrapper with a base net."""
```

Comment:
```suggestion
    """A RewardNet wrapping a base network."""
```

Reply:
Can you also document what is the responsibility of subclasses to implement?

`forward()` definitely. But note by default `predict`, `preprocess`, `predict_processed`, `device` and all the Torch `nn.Module` methods will be inherited from `RewardNet` and *not* passed through to the `base`.

I think this actually works out OK most the time. E.g. `device` looks at parameters, but we get parameters of base network as Torch treats it like a submodule. `predict` calls `forward`, which subclass already needs to implement, etc.

But it's a bit fragile. If the base network had overrode `predict` for some reason, that will get lost as we don't forward to `base.predict`. So important users know this behaviour.